client/pruning/main.go

```diff
@@ -0,0 +1,125 @@
+package pruning
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	pruningtypes "github.com/cosmos/cosmos-sdk/pruning/types"
+	"github.com/cosmos/cosmos-sdk/server"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/cosmos-sdk/store/rootmulti"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+)
+
+const FlagAppDBBackend = "app-db-backend"
+
+// PruningCmd prunes the sdk root multi store history versions based on the pruning options
+// specified by command flags.
+func PruningCmd(appCreator servertypes.AppCreator) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "prune",
+		Short: "Prune app history states based on the pruning options specified by flags",
+		Long: `Prune app history states based on the pruning options specified by flags,
+		Pruning options can be provided via the '--pruning' flag or alternatively with '--pruning-keep-recent', and
+		'--pruning-interval' together.
+		
+		For '--pruning' the options are as follows:
+		
+		default: the last 362880 states are kept, pruning at 10 block intervals
+		nothing: all historic states will be saved, nothing will be deleted (i.e. archiving node)
+		everything: 2 latest states will be kept; pruning at 10 block intervals.
+		custom: allow pruning options to be manually specified through 'pruning-keep-recent', and 'pruning-interval'.
+
+		besides pruning options, database home directory and database backend type should also be specified via flags
+		'--home' and '--app-db-backend'.
+		valid app-db-backend type includes 'goleveldb', 'cleveldb', 'rocksdb', 'boltdb', and 'badgerdb'.
+		`,
+		Example: "prune --home './' --app-db-backend 'goleveldb' --pruning 'custom' --pruning-keep-recent 100 --pruning-interval 10",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			vp := viper.New()
+
+			// Bind flags to the Context's Viper so we can get pruning options.
+			if err := vp.BindPFlags(cmd.Flags()); err != nil {
+				return err
+			}
+			pruningOptions, err := server.GetPruningOptionsFromFlags(vp)
+			if err != nil {
+				return err
+			}
+
+			home := vp.GetString(flags.FlagHome)
+			db, err := openDB(home, server.GetAppDBBackend(vp))
+			if err != nil {
+				return err
+			}
+
+			logger := log.NewTMLogger(log.NewSyncWriter(os.Stdout))
+			app := appCreator(logger, db, nil, vp)
+			cms := app.CommitMultiStore()
+
+			if rootMultiStore, ok := cms.(*rootmulti.Store); ok {
```

Comment:
```suggestion
			rootMultiStore, ok := cms.(*rootmulti.Store)
			if !ok {
			    return err
			}
			happy path
```

This code structure would be better?

Reply:
optimized